src/SignalR/clients/cpp/src/signalrclient/websocket_transport.cpp

```diff
@@ -66,44 +160,37 @@ namespace signalr
             }
 
             pplx::cancellation_token_source receive_loop_cts;
-            pplx::task_completion_event<void> connect_tce;
 
             auto transport = shared_from_this();
 
-            websocket_client->connect(url)
-                .then([transport, connect_tce, receive_loop_cts](pplx::task<void> connect_task)
+            websocket_client->start(url, format, [transport, receive_loop_cts, callback](std::exception_ptr exception)
                 {
                     try
                     {
-                        connect_task.get();
+                        if (exception != nullptr)
+                        {
+                            std::rethrow_exception(exception);
```

Comment:
Why not just call the callback immediately with the exception and return?

Reply:
So we can log the exception